operator/builtin/input/file/config.go

```diff
@@ -117,6 +119,24 @@ func (c InputConfig) Build(context operator.BuildContext) ([]operator.Operator,
 		return nil, fmt.Errorf("invalid start_at location '%s'", c.StartAt)
 	}
 
+	var labelRegex *regexp.Regexp
+	if c.LabelRegex != "" {
+		r, err := regexp.Compile(c.LabelRegex)
+		if err != nil {
+			return nil, fmt.Errorf("compiling regex: %s", err)
+		}
+
+		keys := r.SubexpNames()
+		// keys[0] is always the empty string
+		if len(keys) != 3 {
+			return nil, fmt.Errorf("label_regex must contain two capture groups named 'key' and 'value'")
```

Comment:
What's the third key?

Reply:
The first key is not used, so we expect a regex with two capture groups to be of length 3. You can see this in action here: https://github.com/observIQ/stanza/blob/master/operator/builtin/parser/regex/regex.go#L105.

This behavior is documented here: https://pkg.go.dev/regexp#Regexp.SubexpNames